codegen/aws/core/src/main/java/software/amazon/smithy/python/aws/codegen/AwsDocConverter.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.smithy.python.aws.codegen;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.nodes.Node;
+import org.jsoup.nodes.TextNode;
+import org.jsoup.select.NodeVisitor;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.traits.DocumentationTrait;
+import software.amazon.smithy.model.transform.ModelTransformer;
+import software.amazon.smithy.python.codegen.PythonSettings;
+import software.amazon.smithy.python.codegen.integrations.PythonIntegration;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Add a runtime plugin to convert the HTML docs that are provided by services into
+ * RST
+ */
+@SmithyInternalApi
+public class AwsDocConverter implements PythonIntegration {
+    @Override
+    public Model preprocessModel(Model model, PythonSettings settings) {
+        return ModelTransformer.create().mapShapes(model, shape -> {
+            if (shape.hasTrait(DocumentationTrait.class)) {
+                DocumentationTrait docTrait = shape.getTrait(DocumentationTrait.class).get();
+                String html = docTrait.getValue();
+                String rst = convertHtmlToRst(html);
+                DocumentationTrait newDocTrait = new DocumentationTrait(rst);
```

Comment:
I'd rather run this in `PythonWrtier` where we write documentation. The format of the documentation trait is commonmark, so putting RST in there isn't really correct.

Speaking of commonmark, you need to account for that. The easiest way is to just render the whole thing to HTML like [this](https://github.com/aws/smithy-go/blob/main/codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/DocumentationConverter.java#L73). commonmark can contain HTML (and most of the docs we get from AWS are just HTML) but there's more to it.